presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ColumnValidator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.verifier.checksum;
+
+import com.facebook.presto.sql.tree.Identifier;
+import com.facebook.presto.sql.tree.SingleColumn;
+import com.facebook.presto.verifier.framework.Column;
+
+import java.util.List;
+
+abstract class ColumnValidator
+{
+    abstract List<SingleColumn> generateChecksumColumns(Column column);
```

Comment:
nit `generateValidationColumns`. Not all the validations are checksums.

Reply:
Address here: https://github.com/prestodb/presto/pull/12489#issuecomment-475096230